src/movies.js

```diff
@@ -1,16 +1,113 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors (movies) {
+    const allDirectors = movies.map(function (movie) {
+        return movie.director;
+      })
+    return allDirectors
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (movies) {
+    if (movies.length === 0) {
+        return 0;
+    }
+    let counter = 0;
+    const directorSpielberg = movies.filter(function (movie) {
+        if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')){
+            return counter++;
+        }
+    })
+    return counter
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage (movies) {
+    if (movies.length === 0) {
+        return 0;
+    }
+
+    const averageRating = movies.reduce(function (acc, val){
+        if (!val.rate){
+             return acc;
+             } 
+        return acc + val.rate;
+    }, 0);
+    let roundedAverage =  averageRating / movies.length;
+    return parseFloat(roundedAverage.toFixed(2));
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate (movies) {
+    const filterDramaMovies = movies.filter(function (movie) {
+        if (movie.genre.includes('Drama')){
+            return true;
+        } else {
+            return false
+        }
+    })
```

Comment:
You got the filter right this time. You could directly return the boolean expression.

`const filterDramaMovies = movies.filter(function (movie) {
        return (movie.genre.includes('Drama')) })`